src/frequenz/dispatch/actor.py

```diff
@@ -0,0 +1,275 @@
+            grpc_channel: The gRPC channel to use for communication with the API.
+            svc_addr: Address of the service to connect to.
+            updated_dispatch_sender: A sender for new or updated dispatches.
+            ready_dispatch_sender: A sender for ready dispatches.
+            poll_interval: The interval to poll the API for dispatche changes.
+        """
+        super().__init__(name="dispatch")
+
+        self._client = Client(grpc_channel, svc_addr)
+        self._dispatches: dict[int, Dispatch] = {}
+        self._scheduled: dict[int, asyncio.Task[None]] = {}
+        self._microgrid_id = microgrid_id
+        self._updated_dispatch_sender = updated_dispatch_sender
+        self._ready_dispatch_sender = ready_dispatch_sender
+        self._poll_interval = poll_interval
+
+    async def _run(self) -> None:
+        """Run the actor."""
+        try:
+            while True:
+                await self._fetch()
+                await asyncio.sleep(self._poll_interval.total_seconds())
+        except asyncio.CancelledError:
+            for task in self._scheduled.values():
+                task.cancel()
+            raise
+
+    async def _fetch(self) -> None:
+        """Fetch all relevant dispatches."""
+        old_dispatches = self._dispatches
+        self._dispatches = {}
+
+        try:
+            _logger.info("Fetching dispatches for microgrid %s", self._microgrid_id)
+            async for dispatch in self._client.list(microgrid_id=self._microgrid_id):
+                self._dispatches[dispatch.id] = dispatch
+
+                old_dispatch = old_dispatches.pop(dispatch.id, None)
+                if not old_dispatch:
+                    self._update_dispatch_schedule(dispatch, None)
+                    _logger.info("New dispatch: %s", dispatch)
+                    await self._updated_dispatch_sender.send(Created(dispatch=dispatch))
+                elif dispatch.update_time != old_dispatch.update_time:
+                    self._update_dispatch_schedule(dispatch, old_dispatch)
+                    _logger.info("Updated dispatch: %s", dispatch)
+                    await self._updated_dispatch_sender.send(Updated(dispatch=dispatch))
+
+        except grpc.aio.AioRpcError as error:
+            _logger.error("Error fetching dispatches: %s", error)
+            self._dispatches = old_dispatches
+            return
+
+        for dispatch in old_dispatches.values():
+            _logger.info("Deleted dispatch: %s", dispatch)
+            await self._updated_dispatch_sender.send(Deleted(dispatch=dispatch))
+            if task := self._scheduled.pop(dispatch.id, None):
+                task.cancel()
+
+    def _update_dispatch_schedule(
+        self, dispatch: Dispatch, old_dispatch: Dispatch | None
+    ) -> None:
+        """Update the schedule for a dispatch.
+
+        Schedules, reschedules or cancels the dispatch based on the start_time
```

Comment:
Could you clarify how this happens in this doc?

For example, "- If the start time of the new dispatch is different from the old one, the old one is cancelled."